platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/ProblemSummarizer.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems.internal;
+
+import org.gradle.internal.operations.OperationIdentifier;
+
+import javax.annotation.Nullable;
+
+public interface ProblemSummarizer {
```

Comment:
❌ I remember we discussed that we'll introduce the summarization via delegation. Instead of a new interface, we should have a `ThresholdAwareProblemEmitter implements ProblemEmitter` that delegates to the existing emitter if the threshold value hasn't been reached.

Reply:
That is correct.
I saw some issues with that approach. There is a list of `ProblemEmitter`s used to distribute to the event processing and the problems report. When the `ProblemsSummarizer` also  implements `ProblemsEmitter` it would also end up in that list. So I opted for a separate Interface to be able to guide the injection more precisely.